app/api/probe_dock/reports_api.rb

```diff
@@ -73,17 +73,43 @@ def serialization_options reports
 
           if params[:after]
             ref = TestReport.select('id, created_at').where(api_id: params[:after].to_s).first!
-            rel = rel.where 'created_at > ?', ref.created_at
+            rel = rel.where 'test_reports.created_at > ?', ref.created_at
           end
 
+          rel = rel.joins(:runners) if array_param?(:runnerIds)
+          rel = rel.joins(:projects) if array_param?(:projectIds)
+          rel = rel.joins(test_payloads: :project_version) if array_param?(:projectVersionIds) || array_param?(:projectVersionNames) || params[:projectId].present?
```

Comment:
If I'm not mistaken, you'll be missing the `:projects` join if you're only using the `projectId` and not the `projectIds` query parameter. Also, if you're using both, you might have a double join on the projects table. To avoid the double join, you could do something like this:

```
if %i(projectIds projectVersionIds projectVersionNames).any?{ |param| array_param?(param) } || params[:projectId].present?
  project_version_join = { test_payloads: :project_version }
  if params[:projectId].present? || array_param?(:projectIds)
    project_version_join[:test_payloads] = { project_version: :project }
  end

  rel = rel.joins project_version_join
end